tests/LinAlg/vectorTest.hpp

```diff
@@ -313,6 +320,9 @@ class VectorTests : public TestBase
     real_type* todata = getLocalData(&to);
     v.copyTo(todata);
 
+    // Since we have copied into to's data_host, we need to copy to Device
```

Comment:
Is this extra copy desirable? The test was failing before I added this since the `todata` was not being copied onto the device, and the version of `verifyAnswer` that I implemented calls `copyFromDev`.

The other solution that is possible is just calling `verifyAnswer` on `todata`, but this would require creating an extra `verifyAnswer` implementation for the data type double*. This is possible, but I did not think that it would be a necessary change given the extra copy is a simpler fix.